crates/katana/primitives/src/utils/transaction.rs

```diff
@@ -382,4 +382,228 @@ mod tests {
+            sender_address,
+            class_hash,
+            max_fee.to_u128().unwrap(),
+            chain_id,
+            nonce,
+            false,
+        );
+    
+        assert_eq!(actual_hash, expected_hash);
+    }
+
+    #[test]
+    fn test_compute_declare_v2_tx_hash() {
+        // Starknet mainnet tx hash: https://voyager.online/tx/0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9
+        let expected_hash = felt!("0x836d1c53ecf839a36e1e0bc3a3b9bb8087ca152313da63c3198773c8004cb9");
+
+        let sender_address = felt!("0x020c398d72af5efa4b63f5e3d5ad21e981d6af5f5929cfd2ab0d759ff935be53");
+        let class_hash = felt!("0x0311b6f080fd3385e7154ca3a8568eb7d6aebcb7ff627c1f5e7d2cc99aeb7741");
+        let max_fee = felt!("0x108ae97efa9f8");
+        let chain_id = chain_id::MAINNET;
+        let nonce = felt!("0xb");
+        let compiled_class_hash = felt!("0x311b6f080fd3385e7154ca3a8568eb7d6aebcb7ff627c1f5e7d2cc99aeb7741")
+
+        let actual_hash = compute_declare_v2_tx_hash(
+            sender_address,
+            class_hash,
+            max_fee.to_u128().unwrap(),
+            chain_id,
+            nonce,
+            compiled_class_hash,
+            false,
+        );
+    
+        assert_eq!(actual_hash, expected_hash);
+    }
+
+    #[test]
+    fn test_compute_invoke_v1_tx_hash() {
+        // Starknet mainnet tx hash: https://voyager.online/tx/0x10a50b9fb1b23acadf2624d456578441fd00b94556928bf68478a2b3eabdfe8
+        let expected_hash = felt!("0x10a50b9fb1b23acadf2624d456578441fd00b94556928bf68478a2b3eabdfe8");
+
+        let sender_address = felt!("0x1e8b29765eb24b1cc13e21d9112e1ebebefa7cd5f1aee54be06dc19c831d22");
+        let calldata = vec![
+            felt!("0x2"),
+            felt!("0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7"),
+            felt!("0x219209e083275171774dab1df80982e9df2096516f06319c5c6d71ae0a8480c"),
+            felt!("0x0"),
+            felt!("0x3"),
+            felt!("0x41fd22b238fa21cfcf5dd45a8548974d8263b3a531a60388411c5e230f97023"),
+            felt!("0x3276861cf5e05d6daf8f352cabb47df623eb10c383ab742fcc7abea94d5c5cc"),
+            felt!("0x3"),
+            felt!("0x9"),
+            felt!("0xc"),
+            felt!("0x41fd22b238fa21cfcf5dd45a8548974d8263b3a531a60388411c5e230f97023"),
+            felt!("0x9184e72a000"),
+            felt!("0x0"),
+            felt!("0x9184e72a000"),
+            felt!("0x0"),
+            felt!("0x4634"),
+            felt!("0x0"),
+            felt!("0x2"),
+            felt!("0x49d36570d4e46f48e99674bd3fcc84644ddd6b96f7c741b1562b82f9e004dc7"),
+            felt!("0x53c91253bc9682c04929ca02ed00b3e423f6710d2ee7e0d5ebb06f3ecf368a8"),
+            felt!("0x1e8b29765eb24b1cc13e21d9112e1ebebefa7cd5f1aee54be06dc19c831d22"),
+            felt!("0x646d2c15")
+        ];
+        let max_fee = felt!("0x113b8bbfd40de0");
+        let chain_id = chain_id::MAINNET;
```

Comment:
lets use katana's own [`ChainId`](https://github.com/dojoengine/dojo/blob/27580f6aab337ee31a6b9abeba4e07a001d383af/crates/katana/primitives/src/chain.rs#L71) type.

Reply:
okay. that would be SN_MAIN then
also, do you think i should change chainid in the test written for V1 computation to katana's?

Reply:
yes pls, that'd be nice to keep it consistent

Reply:
okay. you can check it now